daemon/indexing/apollo/index.js

```diff
@@ -4,37 +4,170 @@ var search = require('../lib/search.js')
 var db = require('../lib/db.js')
 
 /*
- * Implentation of a GraphQL server using the Apollo framework.
- * See https://www.apollographql.com/server
+ * Implementation of the Origin GraphQL server.
+  * Uses the Apollo framework: https://www.apollographql.com/server
  *
  */
 
 // Type definitions define the "shape" of the data and specify
 // which ways the data can be fetched from the GraphQL server.
 const typeDefs = gql`
+  ######################
+  #
+  # Query output schema.
+  #
+  ######################
+
+  type User {
+    walletAddr: ID!   # Ethereum wallet address
+    identityAddr: ID  # ERC 725 identity address.
+  }
+
+  type Offer {
+    id: ID!
+    ipfsHash: ID!
+    listingId: ID!
+    buyer: User!
+    # Per contract definition: 1: Created, 2: Accepted, 3: Disputed
+    # NOTE: There is no "Finalized" status stored on-chain but could be useful
+    #       to compute it when resolving this object.
+    status: Int!
+  }
+
+  type Price {
+    currency: String!
+    amount: Float!
+  }
+
+  type Review {
+    ipfsHash: ID!
+    reviewer: User!
+    text: String!
+    rating: Int!
+  }
+
+  # TODO: Add a status indicating if Listing is sold out.
   type Listing {
-    id: String
-    name: String
+    id: ID!
+    ipfsHash: ID!
+    seller: User!
+    title: String!
```

Comment:
Some kind of category should be a part of even our draft work here.

Reply:
Added.